src/core/audio/VorbisConsumer.cpp

```diff
@@ -16,11 +20,9 @@ auto VorbisConsumer::start(const std::string& filename) -> bool {
     sfInfo.format = SF_FORMAT_OGG | SF_FORMAT_VORBIS;
     sfInfo.samplerate = static_cast<int>(this->settings.getAudioSampleRate());
 
-    auto SNDFILE_closer = [](SNDFILE* tag) { sf_close(tag); };
-    std::unique_ptr<SNDFILE, decltype(SNDFILE_closer)> sfFile{sf_open(filename.c_str(), SFM_WRITE, &sfInfo),
-                                                              std::move(SNDFILE_closer)};
+    auto sfFile = audio::make_snd_file(file.native(), SFM_WRITE, &sfInfo);
```

Comment:
This is the wrong place to call `.native()`, right? Should be called inside `make_snd_file` instead.
(Do I understand correctly that implicit conversion will create a copy because there is only an `operator string_type()` but no `operator string_type&()`?)

Sry for only half catching this earlier (:

Reply:
You are right, but you could do this in your PR.